File: 0011-container-with-most-water/0011-container-with-most-water.cpp
class Solution {
public:
    int maxArea(vector<int>& height) {
        int left = 0, right = height.size() - 1;
        int area = INT_MIN;
        while(left < right){
            int narea = min(height[left],height[right]) * (right - left);
            area = max(narea,area);
            if(height[left] <= height[right]){
                left++;
            }
            else{
                right--;
            }
        }
        
        return area;
    }
};

File: 0011-container-with-most-water/0011-container-with-most-water.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        left,right = 0,len(height) - 1

        ans = float('-inf')

        while left < right:
            h = min(height[left],height[right])
            water = h * (right - left)
            ans = max(water,ans)

            # move pointers to avoid duplicates

            if height[left] > height[right]:
                right -= 1
            else:
                left += 1
        
        return ans